Learning paths

DrazenDodik opened the pull request:
- [x] New section for learning paths
- [x] Added fundamentals learning path
- [x] Typeform quiz at the end

noorula reviewed (commented):
Looks good! My two teeny tiny comments:

- In the Azure section on line 16 "Microsoft Azure" -> "Azure". It looks odd mentioning Microsoft at this point as it hasn't been used before in the text. 
- Kubernetes section line 27 "in kube" -> "in Kubernetes" (unless kube is something different and I am just a dum-dum).

DrazenDodik commented:
Alright now I bundled some other changes as a part of this pull request 😅 

- Moved Airflow and Docker tutorials to howto
- Added a really simple Dockerfile to show how to build a custom Docker image (standard base image + requirements.txt)
- Updated 101 stuff to point to the learning path
- Added redirects

noorula reviewed (commented):
Couple of comments:
- Outputs line 14: figured out why this sounded a bit off, "regardless" should be "regardless of whether".
- I think the page order under the Fundamentals section should be different: 
Upload output data should be moved between Download input data and Collect metrics due to the code examples being in this order.
